iree/hal/host/host_semaphore_test.cc

```diff
@@ -0,0 +1,144 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/hal/host/host_semaphore.h"
+
+#include <cstdint>
+#include <thread>  // NOLINT
+
+#include "absl/time/time.h"
+#include "iree/base/status.h"
+#include "iree/base/status_matchers.h"
+#include "iree/testing/gtest.h"
+
+namespace iree {
+namespace hal {
+namespace {
+
+// Tests that a semaphore that is unused properly cleans itself up.
+TEST(HostSemaphoreTest, NoOp) {
+  HostSemaphore semaphore(123u);
+  ASSERT_OK_AND_ASSIGN(uint64_t value, semaphore.Query());
+  EXPECT_EQ(123u, value);
+}
+
+// Tests that a semaphore will accept new values as it is signaled.
+TEST(HostSemaphoreTest, NormalSignaling) {
```

Comment:
Nice tests! Would be nice to pull some of this into the HAL CTS somehow.

Reply:
yeah should be possible - everything here should be valid on all targets, and it'd be nice to have some threading tests too (for host->host waits).